apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabTheorycraft/BuildConstaintCard.tsx

```diff
@@ -0,0 +1,152 @@
+import { StatIcon } from '@genshin-optimizer/gi-svgicons'
+import {
+  CustomNumberInput,
+  CustomNumberInputButtonGroupWrapper,
+  DropdownButton,
+} from '@genshin-optimizer/ui-common'
+import { unit } from '@genshin-optimizer/util'
+import ClearIcon from '@mui/icons-material/Clear'
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  CardHeader,
+  Divider,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  Stack,
+} from '@mui/material'
+import { useContext } from 'react'
+import { ArtifactStatWithUnit } from '../../../../Components/Artifact/ArtifactStatKeyDisplay'
+import CardLight from '../../../../Components/Card/CardLight'
+import type { MinTotalStatKey } from '../../../../Database/DataManagers/CharacterTCData'
+import { minTotalStatKeys } from '../../../../Database/DataManagers/CharacterTCData'
+import { CharTCContext } from './CharTCContext'
+
+export function BuildConstaintCard({ disabled }: { disabled: boolean }) {
+  const {
+    charTC: {
+      optimization: { minTotal },
+    },
+    setCharTC,
+  } = useContext(CharTCContext)
+
+  return (
+    <CardLight>
+      <CardHeader title="Stat Constraints" />
+      <Divider />
+      <Box sx={{ p: 1 }}>
+        <Stack spacing={1}>
+          {Object.entries(minTotal).map(([k, v]) => (
+            <Selector key={k} statKey={k} value={v} disabled={disabled} />
+          ))}
+          <DropdownButton disabled={disabled} title={'Add a Stat Constraint'}>
+            {minTotalStatKeys.map((k) => (
+              <MenuItem
+                key={k}
+                disabled={Object.keys(minTotal).includes(k)}
+                onClick={() =>
+                  setCharTC((charTC) => {
+                    charTC.optimization.minTotal[k] = 0
+                  })
+                }
+              >
+                <ListItemIcon>
+                  <StatIcon statKey={k} />
+                </ListItemIcon>
+                <ListItemText>
+                  <ArtifactStatWithUnit statKey={k} />
+                </ListItemText>
+              </MenuItem>
+            ))}
+          </DropdownButton>
+        </Stack>
+      </Box>
+    </CardLight>
+  )
+}
+
+function Selector({
+  statKey,
+  value,
+  disabled,
+}: {
+  statKey?: MinTotalStatKey
+  value?: number
+  disabled: boolean
+}) {
+  const { setCharTC } = useContext(CharTCContext)
+  const unitStr = unit(statKey)
+  return (
+    <ButtonGroup size="small">
+      <DropdownButton
```

Comment:
https://github.com/frzyc/genshin-optimizer/blob/2d83cf45b38d01b25134b10b2694bb05492d81ce/apps/frontend/src/app/Components/StatEditorList.tsx

Might be easier and more consistent to try to use a StatEditorList?